drivers/gles2/shader_compiler_gles2.cpp

```diff
@@ -754,12 +754,9 @@ String ShaderCompilerGLES2::_dump_node_code(SL::Node *p_node, int p_level, Gener
 				} break;
 
 				case SL::OP_MOD: {
-
-					code += "mod(float(";
-					code += _dump_node_code(op_node->arguments[0], p_level, r_gen_code, p_actions, p_default_actions, p_assigning);
-					code += "), float(";
-					code += _dump_node_code(op_node->arguments[1], p_level, r_gen_code, p_actions, p_default_actions, p_assigning);
-					code += "))";
+					String a = _dump_node_code(op_node->arguments[0], p_level, r_gen_code, p_actions, p_default_actions, p_assigning);
+					String n = _dump_node_code(op_node->arguments[1], p_level, r_gen_code, p_actions, p_default_actions, p_assigning);
+					code += "(" + a + " - " + n + " * (" + a + " / " + n + "))";
```

Comment:
How does it behave when `n == 0`?

Reply:
In C/C++ integer modulo with 0 results in SIGFPE. So, undefined behavior?

Reply:
On my computer GLES3 a % 0 always gives 0. This PR always gives a.